pkg/app/cockroachdb.go

```diff
@@ -0,0 +1,303 @@
+	createCrtDir := []string{"sh", "-c", createCrtDirCmd}
+	_, stderr, err := c.execCommand(ctx, createCrtDir)
+	if err != nil {
+		return false, errors.Wrapf(err, "Error while Creating Cert Directory %s", stderr)
+	}
+
+	createCaCrtCmd := fmt.Sprintf("echo '%s' >> /cockroach/cockroach-client-certs/ca.crt", c.cacrt)
+	createCaCrt := []string{"sh", "-c", createCaCrtCmd}
+	_, stderr, err = c.execCommand(ctx, createCaCrt)
+	if err != nil {
+		return false, errors.Wrapf(err, "Error while Creating ca.crt %s", stderr)
+	}
+
+	createTlsCrtCmd := fmt.Sprintf("echo '%s'>> /cockroach/cockroach-client-certs/client.root.crt", c.tlscrt)
+	createTlsCrt := []string{"sh", "-c", createTlsCrtCmd}
+	_, stderr, err = c.execCommand(ctx, createTlsCrt)
+	if err != nil {
+		return false, errors.Wrapf(err, "Error while Creating tls.crt %s", stderr)
+	}
+
+	createTlsKeyCmd := fmt.Sprintf("echo '%s' >> /cockroach/cockroach-client-certs/client.root.key", c.tlskey)
+	createTlsKey := []string{"sh", "-c", createTlsKeyCmd}
+	_, stderr, err = c.execCommand(ctx, createTlsKey)
+	if err != nil {
+		return false, errors.Wrapf(err, "Error while Creating tls.key %s", stderr)
+	}
+
+	changeFilePermCmd := "cd /cockroach/cockroach-client-certs/ && chmod 0600 *"
+	changeFilePerm := []string{"sh", "-c", changeFilePermCmd}
+	_, stderr, err = c.execCommand(ctx, changeFilePerm)
+	if err != nil {
+		return false, errors.Wrapf(err, "Error while changing certificate file permissions %s", stderr)
+	}
+
+	changeDefaultGCTimeCmd := "./cockroach sql --certs-dir=/cockroach/cockroach-client-certs -e 'ALTER RANGE default CONFIGURE ZONE USING gc.ttlseconds = 10;'"
+	changeDefaultGCTime := []string{"sh", "-c", changeDefaultGCTimeCmd}
+	_, stderr, err = c.execCommand(ctx, changeDefaultGCTime)
+	if err != nil {
+		return false, errors.Wrapf(err, "Error while setting up Garbage Collection time %s", stderr)
+	}
+
+	//time.Sleep(30 * time.Minute)
+	return err == nil, err
+}
+
+func (c *CockroachDB) Object() crv1alpha1.ObjectReference {
+	return crv1alpha1.ObjectReference{
+		Kind:      "statefulset",
+		Name:      c.chart.Release,
+		Namespace: c.namespace,
+	}
+}
+
+func (c *CockroachDB) Uninstall(ctx context.Context) error {
+	cli, err := helm.NewCliClient()
+	if err != nil {
+		return errors.Wrap(err, "failed to create helm client")
+	}
+	err = cli.Uninstall(ctx, c.chart.Release, c.namespace)
+	if err != nil {
+		log.WithError(err).Print("Failed to uninstall app, you will have to uninstall it manually.", field.M{"app": c.name})
+		return err
+	}
+	log.Print("Uninstalled application.", field.M{"app": c.name})
+
+	return nil
+}
+
+func (c *CockroachDB) GetClusterScopedResources(context.Context) []crv1alpha1.ObjectReference {
+	return nil
+}
+
+func (c *CockroachDB) Ping(ctx context.Context) error {
+	log.Print("Pinging the cockroachdb database.", field.M{"app": c.name})
+
+	loginCockroachCmd := fmt.Sprintf("./cockroach sql --certs-dir=/cockroach/cockroach-client-certs --host=%s-public", c.chart.Release)
```

Comment:
I think we can remove Cockroach from these variable names, because we know we are working with cockroach db.